File: CSharp/SpaceVIL/SpaceVIL/Core/IPosition.cs

namespace SpaceVIL.Core
{
    /// <summary>
    /// An interface that describes the attributes of the position of a shape.
    /// <para/> This interface is part of SpaceVIL.Core.IBaseItem.
    /// </summary>
    public interface IPosition
    {
        /// <summary>
        /// Method for setting X coordinate of the left-top corner of a shape.
        /// </summary>
        /// <param name="x">X coordinate.</param>
        void SetX(int x);
        /// <summary>
        /// Method for getting X coordinate of the left-top corner of a shape.
        /// </summary>
        /// <returns>X coordinate.</returns>
        int GetX();
        /// <summary>
        /// Method for setting Y coordinate of the left-top corner of a shape.
        /// </summary>
        /// <param name="y">Y coordinate.</param>
        void SetY(int y);
        /// <summary>
        /// Method for getting Y coordinate of the left-top corner of a shape.
        /// </summary>
        /// <returns>Y coordinate.</returns>
        int GetY();
    }
}

File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/Core/IPosition.java
package com.spvessel.spacevil.Core;

/**
 * An interface that describes the attributes of the position of a shape.
 * <p> This interface is part of com.spvessel.spacevil.Core.IBaseItem.
 */
public interface IPosition {
    /**
     * Method for setting X coordinate of the left-top corner of a shape.
     * @param x X coordinate.
     */
    public void setX(int x);

    /**
     * Method for getting X coordinate of the left-top corner of a shape.
     * @return X coordinate.
     */
    public int getX();

    /**
     * Method for setting Y coordinate of the left-top corner of a shape.
     * @param y Y coordinate.
     */
    public void setY(int y);

    /**
     * Method for getting Y coordinate of the left-top corner of a shape.
     * @return Y coordinate.
     */
    public int getY();
}